sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaIO.java

```diff
@@ -172,88 +171,20 @@ public Read withInputMetadata(Metadata metadata) {
-      return toBuilder().setQueueMaxPollTime(value).build();
-    }
-
-    /**
-     * Returns a new transform which will operate on the text blocks with the
-     * given minimum text length.
-     */
-    public Read withMinimumTextlength(Integer value) {
-      return toBuilder().setMinimumTextLength(value).build();
-    }
-
-    /**
-     * Returns a new transform which will use the synchronous reader.
-     */
-    public Read withParseSynchronously(Boolean value) {
-      return toBuilder().setParseSynchronously(value).build();
-    }
-
-    /**
-     * Path to Tika configuration resource.
-     */
-    public Read withOptions(TikaOptions options) {
-      checkNotNull(options, "TikaOptions cannot be empty.");
-      Builder builder = toBuilder();
-      builder.setFilepattern(StaticValueProvider.of(options.getInput()))
-             .setQueuePollTime(options.getQueuePollTime())
-             .setQueueMaxPollTime(options.getQueueMaxPollTime())
-             .setMinimumTextLength(options.getMinimumTextLength())
-             .setParseSynchronously(options.getParseSynchronously());
-      if (options.getContentTypeHint() != null) {
-        Metadata metadata = this.getInputMetadata();
-        if (metadata == null) {
-            metadata = new Metadata();
-        }
-        metadata.add(Metadata.CONTENT_TYPE, options.getContentTypeHint());
-        builder.setInputMetadata(metadata);
-      }
-      if (options.getTikaConfigPath() != null) {
-        builder.setTikaConfigPath(StaticValueProvider.of(options.getTikaConfigPath()));
-      }
-      if (Boolean.TRUE.equals(options.getReadOutputMetadata())) {
-        builder.setReadOutputMetadata(options.getReadOutputMetadata());
-      }
-      return builder.build();
-    }
-
     @Override
-    public PCollection<String> expand(PBegin input) {
-      checkNotNull(this.getFilepattern(), "Filepattern cannot be empty.");
-      final Bounded<String> read = org.apache.beam.sdk.io.Read.from(new TikaSource(this));
-      PCollection<String> pcol = input.getPipeline().apply(read);
-      pcol.setCoder(getDefaultOutputCoder());
-      return pcol;
+    public PCollection<ParseResult> expand(PCollection<ReadableFile> input) {
+      return input.apply(ParDo.of(new ParseToStringFn(this)));
     }
 
     @Override
     public void populateDisplayData(DisplayData.Builder builder) {
       super.populateDisplayData(builder);
 
-      String filepatternDisplay = getFilepattern().isAccessible()
-        ? getFilepattern().get() : getFilepattern().toString();
-      builder
-          .addIfNotNull(DisplayData.item("filePattern", filepatternDisplay)
-            .withLabel("File Pattern"));
+      //String filepatternDisplay = getFilepattern().isAccessible()
+      //  ? getFilepattern().get() : getFilepattern().toString();
+      //builder
+      //    .addIfNotNull(DisplayData.item("filePattern", filepatternDisplay)
+      //      .withLabel("File Pattern"));
```

Comment:
Below: does a simple getInputMetadata().toString() give unacceptable output?

Reply:
Done, but added trim() as it returns a trailing space for a single entry map, and this extra space is definitely ignorable :-)